File: service/makeaplan/service/blog_service.go
package service

import (
	"context"

	"google.golang.org/grpc"

	pb "makeaplan.com/api/model"
	"makeaplan.com/api/service"
	"makeaplan.com/dao"
)

type blogService struct {
}

func RegisterBlogService(server *grpc.Server) {
	service.RegisterBlogQueryService(server, service.NewBlogQueryService(&blogService{}))
}

func (g *blogService) QueryAllBlog(ctx context.Context, request *service.QueryAllBlogRequest) (*service.QueryAllBlogReply, error) {
	blgs, err := dao.GetAllBlog(ctx)
	if err != nil {
		return nil, err
	}

	bs := make([]*pb.Blog, 0, len(blgs))
	for _, b := range blgs {
		bs = append(bs, b.AsPBPlan())
	}
	return &service.QueryAllBlogReply{Blogs: bs}, nil
}

File: mobile/lib/service/blog_service.dart
import 'package:makeaplan/api/blog.pb.dart' as pb;
import 'package:makeaplan/api/blog_service.pb.dart' as pb;
import 'package:makeaplan/api/blog_service.pbgrpc.dart' as grpc;

import 'grpc_client.dart';

class BlogService {
  static BlogService _singleton;

  BlogService._internal();

  factory BlogService() {
    if (_singleton == null) {
      _singleton = BlogService._internal();
    }
    return _singleton;
  }

  grpc.BlogQueryClient _client() {
    return grpc.BlogQueryClient(
      GRPCClient().client(),
      options: getCommonCallOptions(),
    );
  }

  Future<List<pb.Blog>> queryAllBlog() async {
    print("BlogService.queryAllBlog");
    final v = await _client().queryAllBlog(pb.QueryAllBlogRequest.create());
    return v.blogs;
  }
}
